examples/gallery/embellishments/scalebar.py

```diff
@@ -0,0 +1,140 @@
+     Choose from vertical **T**\(op), **M**\(iddle), or **B**\(ottom) and
+     horizontal **L**\(eft), **C**\(entre), or **R**\(ight). Lower / upper
+     case **j** / **J** mean inside / outside of the map bounding box.
+   - **n**: Give normalized bounding box coordinates as *nx*\/\ *ny*.
+   - **x**: Give plot coordinates as *y*\/\ *x*.
+
+ - length: **+w**. Give a distance value, and, optionally a distance unit.
+   Choose from **e** (meters), **f** (feet), **k** (kilometers) [Default],
+   **M** (statute miles), **n** (nautical miles), or **u** (US survey feet).
+ - origin: **+c**\ [*slon*/]\ *slat*. Control where on the map the scale bar
+   applies. If **+c** is not given the reference point is used. If only
+   **+c** is appended the middle of the map is used. Note that *slon* is only
+   optional for projections with constant scale along parallels, e.g.,
+   Mercator projection.
+ - justify: **+j**. Set the anchor point. Specify a two-character (order
+   independent) code. Choose from vertical **T**\(op), **M**\(iddle), or
+   **B**\(ottom) and horizontal **L**\(eft), **C**\(entre), or **R**\(ight).
+ - offset: **+o**\ *offset*\|\ *xoffset*/\ *yoffset*. Give either a common
+   shift or individual shifts in x (longitude) and y (latitude) directions.
+ - height: Use :gmt-term:`MAP_SCALE_HEIGHT` via :func:`pygmt.config`.
+ - fancy style: **+f**. Get a scale bar that looks like train tracks.
+ - unit: **+u**. Add the distance unit given via **+w** to the single
+   distance values.
+ - label: **+l**. Add the distance unit given via **+w** as label. Append
+   text to get a customized label instead.
+ - alignment: **+a**. Set the label alignment. Choose from **t**\(op)
+   [Default], **b**\(ottom), **l**\(eft), or **r**\(ight).
+"""
+
+# %%
+import pygmt
+
+# Create a new Figure instance
+fig = pygmt.Figure()
+
+# -----------------------------------------------------------------------------
+# Add a plain scale bar
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M0/0/10c",  # Mercator projection with 10 centimeters width
+    frame=["WSne", "af"],
+    # The scale bar is placed at position (j) MiddleCenter, applies at the
+    # reference point (+c is not given), and represents a length (+w) of 1000
+    # kilometers
+    map_scale="jMC+w1000k",
+)
+
+fig.shift_origin(xshift="+w1c")
+
+# -----------------------------------------------------------------------------
+# Add a fancy scale bar
+fig.basemap(
+    region=[-45, -25, -15, 0],
+    projection="M10c",
+    frame=["wSnE", "af"],
+    # Place the scale bar at position (j) MiddleLeft by using MiddleLeft as
+    # anchor point (+j) with an offset (+o) of 1 centimeter in x direction
+    # (longitude) and 0 centimeters in y direction (latitude)
+    # Use a fancy style (+f) to get a scale bar that looks like train tracks
+    # Add the distance unit (+u) to the single distance values
+    map_scale="jML+jML+o1c/0c+w1000k+f+u",
+)
```

Comment:
Maybe multiple map scales in a single basemap is better? I.e,:
```
fig.basemap(
    region=[-45, -25, -15, 0],
    projection="M0/0/10c",  # Mercator projection with 10 centimeters width
    frame=["WSne", "af"]
)
fig.basemap(map_scale="jMC+w1000k")
fig.basemap(map_scale="jML+jML+o1c/0c+w1000k+f+u")
```
It also will be easier to refactor after we have a `Figure.scalebar` method proposed in #2831.

Reply:
I just tried this. For 1000 kilometers, the scale bars overlap. Thus I changed the width to 500 kilometers.

```python
import pygmt


# Create a new Figure instance
fig = pygmt.Figure()

# Mercator projection with 10 centimeters width
fig.basemap(region=[-45, -25, -15, 0], projection="M0/0/10c", frame=["WSne", "af"])

# -----------------------------------------------------------------------------
# Add a plain scale bar
# It is placed at position (j) MiddleCenter, applies at the reference point
# (+c is not given), and represents a length (+w) of 500 kilometers
fig.basemap(map_scale="jMC+w500k")

# -----------------------------------------------------------------------------
# Add a fancy scale bar
# Place it at position (j) MiddleLeft by using MiddleLeft as anchor point (+j)
# with an offset (+o) of 1 centimeter in x direction (longitude) and 0
# centimeters in y direction (latitude)
# Use a fancy style (+f) to get a scale bar that looks like train tracks
# Add the distance unit (+u) to the single distance values
fig.basemap(map_scale="jMR+jMr+o1c/0c+w500k+f+u")

fig.shift_origin(xshift="w+1c")


fig.basemap(region=[-45, -25, -15, 0], projection="M0/0/10c", frame=["wSne", "af"])

# -----------------------------------------------------------------------------
# Add a thick scale bar
# Adjust the GMT default parameter MAP_SCALE_HEIGHT locally (the change applies
# only to the code within the "with" statement)
with pygmt.config(MAP_SCALE_HEIGHT="20p"):
    fig.basemap(
        # The scale bar applies (+c) at the middle of the map (no location is
        # appended to +c)
        # Without appending text, +l adds the distance unit as label
        map_scale="jMC+c+w500k+f+l",
    )

# -----------------------------------------------------------------------------
# Add a scale bar valid for a specific location
fig.basemap(
    # The scale bar applies (+c) at -7° S
    # Add a customized label by appending text to +l
    map_scale="jBR+o1c/1c+c-7+w500k+f+u+lvalid at 7° S",
)

fig.show()
```
_Output figure_:
![scalebars](https://github.com/GenericMappingTools/pygmt/assets/94163266/def9b58a-23dd-4ecb-ba31-3535a6a9bab7)

Reply:
Usually, we put a scale bar at the corner of the map. So I think in this example, we can have four scale bars at the four corners, but the scalebar locations can be specified using different coordinates (e.g., `jBL`, `g-35/-7`, `n0.8/0.8`).

Reply:
I also thought about combining the two subplots into one figure. After the change in commit https://github.com/GenericMappingTools/pygmt/pull/2822/commits/311cca7d2f9972f14e69c1ce8d99eb44140164a0 the gallery example is significantly shorter and clearer 🙂.